### PR TITLE
[UIK-2906][d3-chart] Removed aria-labelledby from LegendItem with shapes other than 'checkbox'

### PR DESCRIPTION
**Pull request by ksimanov**
Removed aria-labelledby from LegendItem with shapes other than 'checkbox'
Also, made aria-labelledby unique

## Motivation and Context

a11y improvement

## How has this been tested?

manually 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [z] Nice improve.

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly or it's not required.
- [x] Unit tests are not broken.
- [x] I have added changelog note to corresponding `CHANGELOG.md` file with planned publish date.
- [ ] I have added new unit tests on added of fixed functionality.

### CODE REVIEW

File: semcore/d3-chart/src/component/ChartLegend/LegendItem/LegendItem.tsx [hunk 1]
```diff
@@ -13,16 +13,21 @@ import {
...
 } from './LegendItem.type';
 import resolveColorEnhance from '@semcore/utils/lib/enhances/resolveColorEnhance';
+import uniqueIDEnhancement from '@semcore/utils/lib/uniqueID';
 import { PatternSymbol } from '../../../Pattern';
 import { getChartDefaultColorName } from '../../../utils';
 
+const enhance = [resolveColorEnhance(), uniqueIDEnhancement()] as const;
 class LegendItemRoot extends Component<
-  LegendItemProps & { resolveColor: ReturnType<typeof resolveColorEnhance> }
+  LegendItemProps & { resolveColor: ReturnType<typeof resolveColorEnhance> },
```
Comment (ilyabrower) [inline semcore/d3-chart/src/component/ChartLegend/LegendItem/LegendItem.tsx:22]: I think, we could remove now `& { resolveColor: ReturnType<typeof resolveColorEnhance> }`

File: semcore/d3-chart/src/component/ChartLegend/LegendItem/LegendItem.tsx [hunk 2]
```diff
@@ -107,7 +117,7 @@ class LegendItemRoot extends Component<
     const disabled = StaticShapes.includes(shape);
 
     return sstyled(styles)(
-      <SLegendItem render={Flex} disabled={disabled}>
+      <SLegendItem render={Flex} disabled={disabled} __excludeProps={'id'}>
```
Comment (ilyabrower) [inline semcore/d3-chart/src/component/ChartLegend/LegendItem/LegendItem.tsx:120]: `__excludeProps ` should be an array

File: semcore/d3-chart/src/component/ChartLegend/LegendItem/LegendItem.tsx [hunk 3]
```diff
@@ -64,12 +73,13 @@ class LegendItemRoot extends Component<
     };
   }
 
-  getLabelProps(): Omit<LegendItem, 'color'> & IRootComponentProps & { onClick: () => void } {
-    const { id, checked, color, onChangeLegendItem, ...props } = this.asProps;
+  getLabelProps(): Omit<LegendItem, 'color' | 'id'> &
```
Comment (ilyabrower) [inline semcore/d3-chart/src/component/ChartLegend/LegendItem/LegendItem.tsx:76]: I'm not sure, that you need to Omit id...

File: semcore/d3-chart/src/component/ChartLegend/LegendItem/LegendItem.tsx [hunk 4]
```diff
@@ -36,10 +36,14 @@ class LegendItemRoot extends Component<
     ),
   });
 
+  getUniqueID() {
+    const { uid, shape } = this.asProps;
+    return shape === 'Checkbox' ? `chart-legend-item-${uid}` : undefined;
```
Comment (ilyabrower) [inline semcore/d3-chart/src/component/ChartLegend/LegendItem/LegendItem.tsx:41]:
```suggestion
    return `chart-legend-item-${uid}`;
```
let's return some id always and pass of not it to the shape in getShapeProps

File: semcore/d3-chart/src/component/ChartLegend/LegendItem/LegendItem.tsx [hunk 5]
```diff
@@ -51,7 +55,7 @@ class LegendItemRoot extends Component<
       onChange: (value: boolean) => {
         onChangeLegendItem(id, value);
       },
-      'aria-labelledby': id,
+      'aria-labelledby': this.getUniqueID(),
```
Comment (ilyabrower) [inline semcore/d3-chart/src/component/ChartLegend/LegendItem/LegendItem.tsx:58]:
```suggestion
      'aria-labelledby': shape === 'Checkbox' ? this.getUniqueID() : undefined,
```

**Review by ilyabrower — APPROVED** [review]

**Review by Valeria-Zimnitskaya — APPROVED** [review]

**Pull request merged**